### ISSUE TITLE
Sharding spec tracker

### ISSUE THREAD

**Issue by protolambda** [ethereum/eth2.0-specs#2546]
The Sharding spec is still **experimental**. This issue tracks the development progress.

TODO:
- [ ] reward header includer (beacon proposer) when header is confirmed
  - add to epoch processing: for each slot in previous epoch, check if union selector equals `SHARD_WORK_CONFIRMED`, and reward the includer_index in the `AttestedDataCommitment`
- [ ] reward less to shard-proposer if header is not confirmed (suggestion by Danny)
  - Currently we reward the priority fee immediately. Instead, we could burn it along with the base-fee, and reward to the shard-proposer upon epoch-processing of previous epoch (along with other TODO above)
- [ ] Clarify attesters get rewarded for attesting the confirmed shard-data timely
  - "timely": confirmation was part of the attestation-processing, not part of the very last moment during rotation of previous-epoch data (`process_pending_shard_confirmations`)
  - the Altair `WEIGHT_DENOMINATOR` needs adjustment, but in a compatible way
  - extending `PARTICIPATION_FLAG_WEIGHTS` is too silent, this changes behavior of Altair (in compatible way). Also, we don't override constants in any fork, so modifying it is a bad pattern.
- [ ] Fix attack vector from builders to build multiple blobs in parallel, without sufficient balance for priority fees. Possible solutions:
  - 1 builder balance per shard (problematic with transactions, liquidity, state size)
  - Enforce builder balance buffer to cover N shards (does not cover super high priority fees though)
  - slashable builders, if building on multiple shards at the same time? (introduces builder entry/exit queue problem)
- [ ] replacement data-transactions, i.e. speed up by paying a higher fee (affects p2p propagation conditions)
- [ ] ensure multiple transactions for sequential slots, while unconfirmed, can be paid by the builder. I.e. the parallel payment problem also applies to a single shard while the txs are unconfirmed.

Other questions:
- Builder entry/exit options. Just baked into genesis state for devnets for now. But some ideas:
  - re-use the existing deposit mechanism, with special withdrawal credential, to indicate it's a builder deposit instead of validator deposit. And similarly for withdrawals.
  - introduce new transaction type for beacon-proposers to create builders with their balance? There could be an eth1 contract that buffers funds, and tops up validators with deposits if supplied with a proof that the validator created a builder with certain credentials.
  - wait for EVM beacon-chain integration to mature, make receipts accessible to the beacon-chain block processing, and implement builder-deposits/withdrawals without new SSZ transaction type? (drawback: one-directional synchrony between execution and consensus layer during sync)

**Comment by leolara** [comment]
I am closing this issue because it seems stale. Please, do not hesitate to reopen it if this is a mistake
